### DISCUSSION TITLE
It is possible to only include certain Assets(AvaloniaResource) depending on platform/cpu arch?

### DISCUSSION THREAD

**Discussion by Shivansps** [AvaloniaUI/Avalonia#12704] [Q&A] [answered]
Lets say, just to put an example, the 7zip console app, im going to have one for file for windows, one for mac, and 3 for linux (x86, x64, arm64). I can can keep them all in my asset folder and only extract the one i need, when i needed to use it. But when i publish i use the "produce single file", what means for each platform  i have a lot of extra files included that are never going to be used. Like the Windows version having the linux and mac bins as well.

There is any way that i can select what assets are included for each platform/cpu?

**Comment by stevemonaco** [comment, marked as answer]
By default, the Avalonia templates will add the following to the .csproj to mark everything as an Avalonia resource:
```xml
  <ItemGroup>
    <AvaloniaResource Include="Assets\**" />
  </ItemGroup>
  ```

What you could try is make many `ItemGroup`s with a `Condition` property for what resources can be included. eg.
```xml
  <ItemGroup Condition="$(RuntimeIdentifier.StartsWith('win'))">
    <AvaloniaResource Include="Assets\General\**" />
    <AvaloniaResource Include="Assets\Win\**" />
  </ItemGroup>
```

I haven't tried the approach, but it seems like it should work for the publish step. You'll have to experiment on the folder structure.

**Comment by Shivansps** [comment]
This works well, thank you.
